### DISCUSSION TITLE
Complete shutdown of Adafruit ItsyBitsy nRF52840 after n-able/nimBLE firmware update

### DISCUSSION THREAD

**Discussion by andrevipid** [h2zero/NimBLE-Arduino#830] [Q&A]
Hello I am programming my Adafruit ItsyBitsy nRF52840 Express dev board (https://www.adafruit.com/product/4481), I have run some Blinky (test) codes just to make sure it works fine. I managed to upload the firmware using board selection adafruit and my arduino firmware worked fine with the board, leds on and switching.

But now I am interested to run and upload the firmware nimBLE arduino server.ino into nRF52840, which is part of this board. I am not interested to use adafruit core library because it does not support the performance as much as n-able arduino / nimBLE does. 

I managed to configure the new n-able arduino board (https://h2zero.github.io/n-able-Arduino/package_n-able_boards_index.json) and installed Arm nimBLE package. 

![image](https://github.com/user-attachments/assets/a229fb96-6039-4bc3-9b68-0981a13d9790)

Therefore, I could select the board properly:

First I tried to use Board -> Itsybitsy bRF52840 Express, however was not able to load using programmer STLINK V2. It does not recognize the stlink and forced to use adafruit bootloader. It was unsuccessful. Then secondly I setup the board to generic nRF52840 like below:

![image](https://github.com/user-attachments/assets/5e88453a-5171-4ef2-9baa-eb23d5e1b246)

I could upload the firmware using Stlinkv2 and arduino ide but with this return message from arduino IDE:

```
---------------------------------------
c:/users/poaand29/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .heap by 4 bytes
c:/users/poaand29/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .stack_dummy by 4 bytes
c:/users/poaand29/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .heap by 4 bytes
c:/users/poaand29/appdata/local/arduino15/packages/h2zero/tools/gcc-arm-none-eabi/9.3.1-1/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld.exe: warning: changing start of section .stack_dummy by 4 bytes
Sketch uses 17684 bytes (1%) of program storage space. Maximum is 1032192 bytes.
Global variables use 4988 bytes of dynamic memory.

xPack OpenOCD x86_64 Open On-Chip Debugger 0.11.0+dev (2022-03-25-17:32)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
debug_level: 0

0x4000

nRF52 device has a CTRL-AP dedicated to recover the device from AP lock.
A high level adapter (like a ST-Link) you are currently using cannot access
the CTRL-AP so 'nrf52_recover' command will not work.
Do not enable UICR APPROTECT.

target halted due to debug-request, current mode: Thread 
xPSR: 0x01000000 pc: 0x0001a860 msp: 0x20040000
** Programming Started **
** Programming Finished **
** Verify Started **
** Verified OK **
** Resetting Target **
shutdown command invoked
--------------------------------------------------------------
```

After that my Adafruit ItsyBitsy nRF52840 express dev board completely shutdown, not power led is on, no USB is recognized through COM ports. So I lost all connectivity to it.

What did I manage wrongly ? I suppose I probably vanished the bootloader from adafruit or locked the nRF52840 due to be using stlink maybe ?

I bought a new board and I want to manage to load the code properly to it. Should I upload it through NRF52840 dongle ?

**Comment by h2zero** [comment]
Your device should be fine, try burning the bootloader (none) which will delete everything and then flash again.
